### ISSUE TITLE
Uncaught ArrayIndexOutOfBoundsException on Import from OnTime

### ISSUE THREAD

**Issue by MattJF90** [Rsl1122/Plan-PlayerAnalytics#35]
**Plan Version:** 2.7.2
**OnTime Version:** 4.1.3
**Database Type:** mysql
**Command Causing Issue:** /plan manage import OnTime -a

**Description:**
While importing OnTime player data into Plan, an exception is thrown part way through the import.
After the import, it is no longer possible to execute /plan analyze, as an exception is thrown.

I suspect that the plugin has either imported data from OnTime incorrectly, or OnTime has some data which is not valid (note: OnTime is working correctly).

**Steps to Reproduce:**
1. Execute /plan manage import OnTime -a
2. Execute /plan analyze

**Stack Trace - Console:**
```
02.03 11:48:55 [Server] ERROR [Plan] Ran into an error. It has been logged to the Errors.txt
02.03 11:48:56 [Server] WARN Exception in thread "Craft Scheduler Thread - 72" 
02.03 11:48:56 [Server] WARN org.apache.commons.lang.UnhandledException: Plugin Plan v2.7.2 generated an exception while executing task 25972
02.03 11:48:56 [Server] INFO at org.bukkit.craftbukkit.v1_10_R1.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:56)
02.03 11:48:56 [Server] INFO at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
02.03 11:48:56 [Server] INFO at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
02.03 11:48:56 [Server] INFO at java.lang.Thread.run(Thread.java:745)
02.03 11:48:56 [Server] INFO Caused by: java.lang.ArrayIndexOutOfBoundsException

```

**Stack Trace - Errors.txt:**
```
[Mar 02 10:36:03] main.java.com.djrapitops.plan.data.cache.GetConsumer Caught java.lang.ArrayIndexOutOfBoundsException: 14
[Mar 02 10:36:03]   sun.util.calendar.BaseCalendar.getCalendarDateFromFixedDate(BaseCalendar.java:453)
[Mar 02 10:36:03]   java.util.GregorianCalendar.computeFields(GregorianCalendar.java:2397)
[Mar 02 10:36:03]   java.util.GregorianCalendar.computeFields(GregorianCalendar.java:2312)
[Mar 02 10:36:03]   java.util.Calendar.setTimeInMillis(Calendar.java:1804)
[Mar 02 10:36:03]   java.util.Calendar.setTime(Calendar.java:1770)
[Mar 02 10:36:03]   java.text.SimpleDateFormat.format(SimpleDateFormat.java:943)
[Mar 02 10:36:03]   java.text.SimpleDateFormat.format(SimpleDateFormat.java:936)
[Mar 02 10:36:03]   java.text.DateFormat.format(DateFormat.java:345)
[Mar 02 10:36:03]   net.minecraft.server.v1_10_R1.UserCache$BanEntrySerializer.a(UserCache.java:299)
[Mar 02 10:36:03]   net.minecraft.server.v1_10_R1.UserCache$BanEntrySerializer.serialize(UserCache.java:345)
[Mar 02 10:36:03]   net.minecraft.server.v1_10_R1.UserCache$BanEntrySerializer.serialize(UserCache.java:1)
[Mar 02 10:36:03]   com.google.gson.TreeTypeAdapter.write(TreeTypeAdapter.java:70)
[Mar 02 10:36:03]   com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:68)
[Mar 02 10:36:03]   com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.write(CollectionTypeAdapterFactory.java:96)
[Mar 02 10:36:03]   com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.write(CollectionTypeAdapterFactory.java:60)
[Mar 02 10:36:03]   com.google.gson.Gson.toJson(Gson.java:593)
[Mar 02 10:36:03]   com.google.gson.Gson.toJson(Gson.java:572)
[Mar 02 10:36:03]   com.google.gson.Gson.toJson(Gson.java:527)
[Mar 02 10:36:03]   com.google.gson.Gson.toJson(Gson.java:507)
[Mar 02 10:36:03]   net.minecraft.server.v1_10_R1.UserCache.c(UserCache.java:231)
[Mar 02 10:36:03]   net.minecraft.server.v1_10_R1.UserCache.getProfile(UserCache.java:164)
[Mar 02 10:36:03]   org.bukkit.craftbukkit.v1_10_R1.CraftProfileBanList.isBanned(CraftProfileBanList.java:84)
[Mar 02 10:36:03]   org.bukkit.craftbukkit.v1_10_R1.CraftOfflinePlayer.isBanned(CraftOfflinePlayer.java:97)
[Mar 02 10:36:03]   main.java.com.djrapitops.plan.data.UserData.<init>(UserData.java:94)
[Mar 02 10:36:03]   main.java.com.djrapitops.plan.database.databases.SQLDB.giveUserDataToProcessors(SQLDB.java:507)
[Mar 02 10:36:03]   main.java.com.djrapitops.plan.data.cache.GetConsumer.consume(DataCacheGetQueue.java:72)
[Mar 02 10:36:03]   main.java.com.djrapitops.plan.data.cache.GetConsumer.run(DataCacheGetQueue.java:60)
[Mar 02 10:36:03]   java.lang.Thread.run(Thread.java:745)
```

**Comment by Rsl1122** [comment]
This is extremely odd exception.
It appears that it is caused by some code inside net.minecraft.server code when it is checking whether or not the player is banned. 
I'm afraid I can't do anything but try to ignore all players that this exception happens to.

**Comment by Rsl1122** [comment]
Thank you for a very well formatted issue report. :)

**Comment by MattJF90** [comment]
Thanks for replying.

Either ignoring the affected players, ignoring the isBanned property for the affected players, or giving more information in the exception (such as player name/UUID) would definitely be helpful.

After analyzing the stack trace a bit more, it seems that this is an issue with how the date of the ban is being stored for my server. I'll have to check through my banned players and make sure that they are set up correctly in the meantime.

**Comment by Rsl1122** [comment]
I guess I can set banned to false by catching the exception while setting isBanned.
By doing this all affected player's data will still be imported.

**Comment by Rsl1122** [comment]
Should be fixed in 2.7.3
